resources/js/mp-history/game.tsx

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import StringWithComponent from 'components/string-with-component';
+import BeatmapJson from 'interfaces/beatmap-json';
+import BeatmapsetJson from 'interfaces/beatmapset-json';
+import LegacyMatchEventJson from 'interfaces/legacy-match-event-json';
+import Ruleset from 'interfaces/ruleset';
+import ScoreJson from 'interfaces/score-json';
+import UserJson from 'interfaces/user-json';
+import _ from 'lodash';
+import * as React from 'react';
+import { classWithModifiers } from 'utils/css';
+import { formatNumber } from 'utils/html';
+import { trans, transExists } from 'utils/lang';
+import GameHeader from './game-header';
+import Score from './score';
+
+interface TeamScores {
+  blue: number;
+  red: number;
+}
+
+type SortedScore = ScoreJson & {
+  teamRank: number;
+};
+
+interface Props {
+  event: LegacyMatchEventJson;
+  teamScores: TeamScores;
+  users: Partial<Record<number, UserJson>>;
+}
+
+export default function Game(props: Props) {
+  const game = props.event.game!;
+
+  if (game.scores.some((score) => score.match == null)) {
+    throw new Error('scores are missing match data');
+  }
+
+  const showTeams = game.team_type === 'team-vs' || game.team_type === 'tag-team-vs';
+
+  const winningTeam = props.teamScores.blue > props.teamScores.red ? 'blue' : 'red';
+  const difference = Math.abs(props.teamScores.blue - props.teamScores.red);
+
+  let sortedScores = game.scores.map((m) => {
+    const sortedScore = m as SortedScore;
+    sortedScore.teamRank = m.match!.team === winningTeam ? 1 : 2;
+    return sortedScore;
+  });
+
+  sortedScores = _.orderBy(sortedScores, ['teamRank', 'score'], ['asc', 'desc']);
```

Comment:
using plain sort with relevant callback function will remove the need for the `& { teamRank: ...}` type